platform/broadcom/sonic-platform-modules-mitac/ly1200-32x/opt/xcvr-serv/bms_xcvr_access.c

```diff
@@ -0,0 +1,345 @@
+ {    0,      6,     0,     1 },/* "Initialization complete flag" is bit0 of byte 6 in the page0 */
+ {    0,    129,     0,     8 },/* "Ext. Identifier" is bit0-1 of byte129 in the page0 */
+ {    0,     93,     0,     3 },/* "Power Mode Control" is bit0-2 of byte 93 in the page0 */
+};
+int bms_xcvr_data_list_num = sizeof(bms_xcvr_data_list) / sizeof(bms_xcvr_data_list[0]);
+
+/*
+ * FEATURE:
+ *   bms_xcvr_i2c_open
+ * PURPOSE:
+ *   low-level function to open an I2C adapter device file
+ * PARAMETERS:
+ *   bus                (IN) I2C bus to be looking for
+ *   i2c_fd             (OUT) file descriptor
+ * RETURNS:
+ *   open success, this function returns .
+ *   for other cases, negative errno is returned.
+ */
+static int
+bms_xcvr_i2c_open(unsigned char bus,
+                  int *i2c_fd)
+{
+    int err = 0;
+    char dev_name[20];
+
+    sprintf(dev_name, "/dev/i2c-%u", bus);
+    if ((*i2c_fd = open(dev_name, O_RDWR)) < 0) {
+        err = errno;
+        syslog(LOG_DEBUG|LOG_USER,
+            "[%s] could not open I2C adapter %s: %s\n",
+            __func__, dev_name, strerror(err));
+        return -err;
+    }
+
+    return err;
+}
+
+/*
+ * FEATURE:
+ *   bms_xcvr_i2c_read
+ * PURPOSE:
+ *   low-level function to read 8-bit data from an I2C slave component
+ * PARAMETERS:
+ *   bus                (IN) I2C bus to be looking for
+ *   addr               (IN) I2C slave address of the transceiver
+ *   offset             (IN) register offset
+ *   val                (OUT) pointer of current value buffer
+ * RETURNS:
+ *   read success, this function returns 0.
+ *   for other cases, negative errno is returned.
+ */
+static int
+bms_xcvr_i2c_read(unsigned char bus,
+                  unsigned char addr,
+                  unsigned short offset,
+                  unsigned char *val)
+{
+    int err = 0, value;
+    int i2c_fd = 0;
+
+    if ((err = bms_xcvr_i2c_open(bus, &i2c_fd)) < 0) {goto quit;}
+    if ((ioctl(i2c_fd, I2C_SLAVE_FORCE, addr)) < 0) {
+        err = errno;
+        return -err;
+    }
+    value = i2c_smbus_read_byte_data(i2c_fd, offset);
+    if (value < 0) {
+        err = errno;
+        syslog(LOG_DEBUG|LOG_USER,
+            "[%s] could not read register 0x%X: %s\n",
+            __func__, offset, strerror(err));
+        close(i2c_fd);
+        return -err;
+    }
+    *val = value;
+
+quit:
+    close(i2c_fd);
+    return err;
+}
+
+/*
+ * FEATURE:
+ *   bms_xcvr_i2c_write
+ * PURPOSE:
+ *   low-level function to write 8-bit data to an I2C slave component
+ * PARAMETERS:
+ *   bus                (IN) I2C bus to be looking for
+ *   addr               (IN) I2C slave address of the transceiver
+ *   offset             (IN) register offset
+ *   val                (IN) new register value
+ * RETURNS:
+ *   write success, this function returns 0.
+ *   for other cases, negative errno is returned.
+ */
+static int
+bms_xcvr_i2c_write(unsigned char bus,
```

Comment:
we have sff_8436_eeprom driver allow you user  to access sfp eeprom data via sysfs. It is redudant to have same code in user space that does the same thing.